UI/States/CraftingUIState.cs

```diff
@@ -1205,17 +1205,37 @@ or ButtonConfigurationMode.LegacyWithGear
 						CraftingGUI.SetNextDefaultRecipeCollectionToRefresh(obj.StoredItem.type);
 						CraftingGUI.forceSpecificRecipeResort = true;
 					} else if (MagicStorageConfig.RecipeBlacklistEnabled && Main.keyState.IsKeyDown(Keys.LeftControl)) {
-						if (recipeButtons.Choice == CraftingGUI.RecipeButtonsBlacklistChoice) {
-							if (storagePlayer.HiddenRecipes.Remove(obj.StoredItem)) {
-								Main.NewText(Language.GetTextValue("Mods.MagicStorage.RecipeRevealed", Lang.GetItemNameValue(obj.StoredItem.type)));
+						if (Main.keyState.IsKeyDown(Keys.LeftShift)) {
+							if (recipeButtons.Choice == CraftingGUI.RecipeButtonsBlacklistChoice) {
+								if (MagicStorageConfig.GlobalRecipeBlacklist.Remove(new(obj.StoredItem.type))) {
+									Main.NewText(Language.GetTextValue("Mods.MagicStorage.RecipeRevealedGlobal", Lang.GetItemNameValue(obj.StoredItem.type)));
 
-								slotZone.SetItemsAndContexts(int.MaxValue, GetRecipe);
+									Utility.SaveModConfig(MagicStorageConfig.Instance);
+
+									slotZone.SetItemsAndContexts(int.MaxValue, GetRecipe);
+								}
+							} else {
+								if (MagicStorageConfig.GlobalRecipeBlacklist.Add(new(obj.StoredItem.type))) {
+									Main.NewText(Language.GetTextValue("Mods.MagicStorage.RecipeHiddenGlobal", Lang.GetItemNameValue(obj.StoredItem.type)));
+
+									Utility.SaveModConfig(MagicStorageConfig.Instance);
+
+									slotZone.SetItemsAndContexts(int.MaxValue, GetRecipe);
+								}
 							}
 						} else {
-							if (storagePlayer.HiddenRecipes.Add(obj.StoredItem)) {
-								Main.NewText(Language.GetTextValue("Mods.MagicStorage.RecipeHidden", Lang.GetItemNameValue(obj.StoredItem.type)));
+							if (recipeButtons.Choice == CraftingGUI.RecipeButtonsBlacklistChoice) {
+								if (storagePlayer.HiddenRecipes.Remove(obj.StoredItem)) {
+									Main.NewText(Language.GetTextValue("Mods.MagicStorage.RecipeRevealed", Lang.GetItemNameValue(obj.StoredItem.type)));
 
-								slotZone.SetItemsAndContexts(int.MaxValue, GetRecipe);
+									slotZone.SetItemsAndContexts(int.MaxValue, GetRecipe);
+								}
+							} else {
+								if (storagePlayer.HiddenRecipes.Add(obj.StoredItem)) {
+									Main.NewText(Language.GetTextValue("Mods.MagicStorage.RecipeHidden", Lang.GetItemNameValue(obj.StoredItem.type)));
+
+									slotZone.SetItemsAndContexts(int.MaxValue, GetRecipe);
+								}
 							}
 						}
```

Comment:
This is way to many branches to keep track of.
Add the below code and then use `hideRecipe` and `revealRecipe` (feel free to choose better names) to hide/reveal the recipe
```cs
void RevealPlayerRecipe(Item item)
{
    if (storagePlayer.HiddenRecipes.Remove(item))
    {
        Main.NewText(Language.GetTextValue("Mods.MagicStorage.RecipeRevealed", Lang.GetItemNameValue(obj.StoredItem.type)));
        slotZone.SetItemsAndContexts(int.MaxValue, GetRecipe);
    }
}
void HidePlayerRecipe(Item item)
{
    if (storagePlayer.HiddenRecipes.Add(item))
    {
        Main.NewText(Language.GetTextValue("Mods.MagicStorage.RecipeHidden", Lang.GetItemNameValue(obj.StoredItem.type)));
        slotZone.SetItemsAndContexts(int.MaxValue, GetRecipe);
    }
}
void RevealGlobalRecipe(Item item)
{
    if (MagicStorageConfig.GlobalRecipeBlacklist.Remove(new(item.type)))
    {
        Main.NewText(Language.GetTextValue("Mods.MagicStorage.RecipeRevealedGlobal", Lang.GetItemNameValue(obj.StoredItem.type)));
        Utility.SaveModConfig(MagicStorageConfig.Instance);
        slotZone.SetItemsAndContexts(int.MaxValue, GetRecipe);
    }
}
void HideGlobalRecipe(Item item)
{
    if (MagicStorageConfig.GlobalRecipeBlacklist.Add(new(item.type)))
    {
        Main.NewText(Language.GetTextValue("Mods.MagicStorage.RecipeHiddenGlobal", Lang.GetItemNameValue(obj.StoredItem.type)));
        Utility.SaveModConfig(MagicStorageConfig.Instance);
        slotZone.SetItemsAndContexts(int.MaxValue, GetRecipe);
    }
}

var hideRecipe = Main.keyState.IsKeyDown(Keys.LeftShift) ? HideGlobalRecipe : HidePlayerRecipe;
var revealRecipe = Main.keyState.IsKeyDown(Keys.LeftShift) ? RevealGlobalRecipe : RevealPlayerRecipe;
```

Reply:
fixed